src/cljs/main/broadfcui/utils/user.cljs

```diff
@@ -56,12 +57,18 @@
 
 
 (defonce profile (atom false))
+(defonce terra-preference (atom false)) ;; initially false to disable redirects until profile is loaded
 
 (defn reload-profile [& [on-done]]
   (endpoints/profile-get
    (fn [{:keys [success? get-parsed-response] :as response}]
      (when success?
-       (reset! profile (common/parse-profile (get-parsed-response))))
+       (reset! profile (common/parse-profile (get-parsed-response)))
```

Comment:
Setting and then reading the profile atom feels like an anti-pattern, why not do something like
```cljs
[{:keys [preferTerra] :as profile} (common/parse-profile (get-parsed-response))]
```
and then go from there?

Reply:
@zarsky-broad is my latest commit more palatable? Not the exact syntax you suggested - imho more readable but less  concise.

Reply:
Works for me!